src/prelude/fuse.rs

```diff
@@ -154,16 +155,16 @@ macro_rules! fuse_from_impl(
 
             // Construct all fingerprints (see Algorithm 4 in the paper).
             #[allow(non_snake_case)]
-            let mut B: Box<[$fpty]> = make_block!(with capacity sets);
+            let mut B: Box<[$fpty]> = make_fp_block!(capacity);
             for ki in stack.iter().rev() {
                 let H012 { hset: [h0, h1, h2] } = H012::from(ki.hash, segment_length);
-                let mut fp = fingerprint!(ki.hash) as $fpty;
-                match ki.index {
-                    h if h == h0 => fp ^= B[h1] ^ B[h2],
-                    h if h == h1 => fp ^= B[h0] ^ B[h2],
-                    h if h == h2 => fp ^= B[h0] ^ B[h1],
+                let fp = fingerprint!(ki.hash) as $fpty;
+                let fp = match ki.index {
+                    h if h == h0 => fp ^ B[h1] ^ B[h2],
+                    h if h == h1 => fp ^ B[h0] ^ B[h2],
+                    h if h == h2 => fp ^ B[h0] ^ B[h1],
                     _ => unreachable!(),
```

Comment:
nit: I think you don't need to declare the first fp. I think it may be an awkward style. I would just do
=> fingerprint!(ki.has) as $fpty ^ B[h1] ^ B[h2]